pokemonterminal/main.py

```diff
@@ -110,28 +69,40 @@ def main(argv=None):
         print("Dry run, exiting.")
         return
 
-    if options.clear:
-        if PIPE_EXISTS:
-            pipe_out = os.open(PIPE_PATH, os.O_WRONLY)
-            os.write(pipe_out, b"quit\n")
-            os.close(pipe_out)
-        scripter.clear_terminal()
-        return
-
-    if is_slideshow and options.id <= 0 and size > 1:
-        if os.name == 'nt':
-            print("Slideshow not supported on Windows yet.")
-            sys.exit(0)
-        if PIPE_EXISTS:
-            print("Slideshow already running in this instance!")
-            sys.exit(0)
-        if options.slideshow <= 0:
-            print("Time has to be greater then 0. You can use decimal values.")
+    e = create_named_event("Pokemon-Terminal_Wallpaper" if options.wallpaper else "Pokemon-Terminal_Terminal")
```

Comment:
I think we only want to do this if the slideshow option is provided.
Otherwise, if you just run `pokemon` to set a static background it initializes the semaphore and sets `PosixNamedEvent.__duplicate` to False.
Then, if you run `pokemon -ss` to start the slideshow, it will hit the `try/execpt` in `PosixNamedEvent` constructor, set `PosixNamedEvent.__duplicate` to true, and then tell you a slideshow is already running when it actually isn't

Reply:
Except if there's something I'm not aware of, shouldn't a named semaphore be closed if all handles to it are closed? (which we do with the finally block here)

Reply:
hmm maybe I'm diagnosing the symptoms wrong then.
I'm seeing it always tell me a slideshow is already started, if i try to start one after having set a static background

Reply:
actually it doesn't matter if i've set a static background or not, it just always tells me a slideshow is already started when i try to start one

Reply:
You probably have a background slideshow process still hanging around (they don't die when their parent terminal exits, not sure how to gracefully detect this situation - maybe check if stdin/stdout actually works or something)

Reply:
I can normally see (and kill) them from `ps` but I see no python processes hanging around

Reply:
```

Reply:
which is what the create flags are for in the constructor.

What happens currently is if you run any `pokemon` command, the semaphore gets created and it outlives all of the processes. This is the reason if you then try to run `pokemon -ss` it says a slideshow has already been started: the semaphore still exists from the last `pokemon` command run, no matter what command it was or when it was run

Reply:
If you unlink() (for example you put that in the close() function of PosixNamedEvent) the semaphore in main, the child will still hold a handle to the semaphore, but new opens will refer to a new semaphore, making duplicate detection not working.

Reply:
i know. But  what you have now is also not correct.
You could do something gross like unlink if slideshow isn't specified, but that's getting messy..

Reply:
We could this in the slideshow worker before exit, altho it feels kinda hacky, since we're ignoring the abstraction we set up.

```py
    finally:
        e.close()
        if sys.platform != "win32":
            e.unlink()

Reply:
Doing anything in the slideshow worker won't fix it.
starting fresh:
`$ pokemon`
Now I have a static background set, and I've created the semaphore.
Now
`$ pokemon -ss` will try to create the semaphore again, hit the `try/except` andset `__is_duplicate`. This will make it say a slideshow is already running when one isn't and one never was

Reply:
Right. A boolean flag `needs_unlink` for main would work, and we should keep e.unlink() in the worker too.

Reply:
Maybe just use a FIFO?

Reply:
Yeah tbh with the issues that @samosaara is also seeing I'm wodering if we should go back to the named FIFO for posix, wrapped in the `NamedEvent` interface

Reply:
See my edit on previous comment, doesn't fixes the issue of having to delete the actual FIFO somewhere and we could set up a new method to check for mutex (with OpenMutex) and semaphore (with the constructor and `flags=0`) existance, bringing them pretty much par to FIFOs)

Reply:
What was done before was deleting the FIFO was part of the clean up of the slideshow task when it was terminated.
Anyway,
if we can check if the semaphore exists, IMO this is what we should be doing:
`pokemon -ss` should be the only thing to create the semaphore
`pokemon -c` should check if the semaphore exists, and if so do what's needed to kill the background process, otherwise just clear screen

Reply:
Sounds good to me. I'll add that tomorrow afternoon.

Reply:
Done and pushed, should work now (I hope)